packages/amplify-graphql-api-construct/src/__tests__/internal/imported-tables.test.ts

```diff
@@ -0,0 +1,125 @@
+      const definition = AmplifyGraphqlDefinition.fromString(
+        /* GraphQL */ `
+          type Todo @model {
+            description: String!
+          }
+        `,
+        {
+          dbType: 'DYNAMODB',
+          provisionStrategy: 'IMPORTED_AMPLIFY_TABLE',
+        },
+      );
+      const tableMap = {
+        Todo: 'Todo-123-dev',
+      };
+      expect(() => validateImportedTableMap(definition, tableMap)).not.toThrow();
+    });
+
+    test('errors when model is missing in map', () => {
+      const definition = AmplifyGraphqlDefinition.fromString(
+        /* GraphQL */ `
+          type Todo @model {
+            description: String!
+          }
+
+          type NotTodo @model {
+            description: String!
+          }
+
+          type Bar @model {
+            description: String!
+          }
+        `,
+        {
+          dbType: 'DYNAMODB',
+          provisionStrategy: 'IMPORTED_AMPLIFY_TABLE',
+        },
+      );
+      const tableMap = {
+        Todo: 'Todo-123-dev',
+      };
+      expect(() => validateImportedTableMap(definition, tableMap)).toThrow(
+        'Cannot find imported Amplify DynamoDB table mapping for models NotTodo, Bar.',
+      );
+    });
+
+    test('errors when model is missing in schema', () => {
+      const definition = AmplifyGraphqlDefinition.fromString(
+        /* GraphQL */ `
+          type Todo @model {
+            description: String!
+          }
+        `,
+        {
+          dbType: 'DYNAMODB',
+          provisionStrategy: 'IMPORTED_AMPLIFY_TABLE',
+        },
+      );
+      const tableMap = {
+        Todo: 'Todo-123-dev',
+        NotTodo: 'NotTodo-123-dev',
+        Bar: 'Bar-123-dev',
+      };
+      expect(() => validateImportedTableMap(definition, tableMap)).toThrow(
+        'Table mapping includes tables not specified as imported in schema. (NotTodo, Bar)',
+      );
+    });
+
+    test('errors on empty map with imported strategy', () => {
+      const definition = AmplifyGraphqlDefinition.fromString(
+        /* GraphQL */ `
+          type Todo @model {
+            description: String!
+          }
+        `,
+        {
+          dbType: 'DYNAMODB',
+          provisionStrategy: 'IMPORTED_AMPLIFY_TABLE',
+        },
+      );
+      const tableMap = {};
+      expect(() => validateImportedTableMap(definition, tableMap)).toThrow(
+        'Cannot find imported Amplify DynamoDB table mapping for models Todo.',
+      );
+    });
+
+    test('errors on missing map with imported strategy', () => {
+      const definition = AmplifyGraphqlDefinition.fromString(
+        /* GraphQL */ `
+          type Todo @model {
+            description: String!
+          }
+        `,
+        {
+          dbType: 'DYNAMODB',
+          provisionStrategy: 'IMPORTED_AMPLIFY_TABLE',
+        },
+      );
+      expect(() => validateImportedTableMap(definition)).toThrow('Table mapping is missing for imported Amplify DynamoDB table strategy.');
+    });
+  });
+});
```

Comment:
We should also have multi-strategy tests. One case should ensure that a model that uses a different strategy throws an error if it is included in the `importedAmplifyDynamoDBTableMap`. Ideally, the error would note that the strategy is invalid, rather than a generic "missing" or "invalid" error. E.g., from your generator test below:

```typescript
dataSourceStrategies: {
        Comment: DDB_DEFAULT_DATASOURCE_STRATEGY,
        Post: DDB_AMPLIFY_MANAGED_DATASOURCE_STRATEGY,
        Author: IMPORTED_DDB_AMPLIFY_MANAGED_DATASOURCE_STRATEGY,
      },
      importedAmplifyDynamoDBTableMap: {
        Author: 'Author-myApiId-myEnv',
      },
```

If you add `Post` to the map, we'd hope to see an error like `"Table mapping includes models that do not use an imported strategy (Post)."` or similar